tempodb/encoding/vparquet4/block.go

```diff
@@ -33,3 +38,40 @@ func newBackendBlock(meta *backend.BlockMeta, r backend.Reader) *backendBlock {
 func (b *backendBlock) BlockMeta() *backend.BlockMeta {
 	return b.meta
 }
+
+// Validate will do a basic sanity check of the state of the parquet file. This can be extended to do more checks in the future.
+// This method should lean towards being cost effective over complete.
+func (b *backendBlock) Validate(ctx context.Context) error {
+	if b.meta == nil {
```

Comment:
Is there a way to reassemble this information from data on disk and still avoid paying the allocs on startup like you mention above?

Reply:
possibly? what do you mean by "this information"?